quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -1284,9 +1294,110 @@ impl MetastoreService for PostgresqlMetastore {
 
     async fn open_shards(
         &mut self,
-        _request: OpenShardsRequest,
+        request: OpenShardsRequest,
     ) -> MetastoreResult<OpenShardsResponse> {
-        unimplemented!("`open_shards` is not implemented for PostgreSQL metastore")
+        let mut subresponses = Vec::with_capacity(request.subrequests.len());
+
+        let grouped_requests: HashMap<IndexUid, Vec<OpenShardsSubrequest>> = request
```

Comment:
The group by thing does not make sense for the Pg implementation. See this comment in the file-backed metastore implementation:
> We must group the subrequests by `index_uid` to mutate each index only once since each mutation triggers an IO.